packages/blockchain-wallet-v4/src/signer/eth.js

```diff
@@ -14,24 +14,30 @@ const toHex = (value) => {
 }
 
 export const signErc20 = curry((network = 1, mnemonic, txnData, contractAddress) => {
-  const { amount, data, gasLimit, gasPrice, index, nonce, to } = txnData
+  const { amount, depositAddress, gasLimit, gasPrice, index, nonce, to } = txnData
   const wallet = ethers.Wallet.fromMnemonic(mnemonic, `m/44'/60'/0'/0/${index}`)
   const transferMethodHex = '0xa9059cbb'
 
   // block ERC20 transfers/sends that are being created with 0 amount
   if (new BigNumber(amount).isZero()) {
     return Task.rejected(new Error('erc20_amount_cannot_be_zero'))
   }
-
+  let data =
+    transferMethodHex +
+    ethers.utils.defaultAbiCoder
+      .encode(['address', 'uint256'], [to, amount.toString()])
+      .replace('0x', '')
+  let gasLimitAdjusted = gasLimit
+  // If a deposit address is specified, append the the address to the end of data payload and increase the gasLimit.
+  if (depositAddress) {
+    data += depositAddress.replace('0x', '')
+    gasLimitAdjusted += 600
```

Comment:
(minor): we might move 600 to be in some const so can be reuse later on line 52, and easier change in future

Reply:
Good catch, thanks! Have remedied that now 👍